exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/impl/MurmurHash3.java

```diff
@@ -0,0 +1,264 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+
+package org.apache.drill.exec.expr.fn.impl;
+
+import io.netty.buffer.DrillBuf;
+import io.netty.util.internal.PlatformDependent;
+
+
+/**
+ *
+ * MurmurHash3 was written by Austin Appleby, and is placed in the public
+ * domain.
+ * See http://smhasher.googlecode.com/svn/trunk/MurmurHash3.cpp
+ * MurmurHash3_x64_128
+ * MurmurHash3_x86_32
+ */
+public final class MurmurHash3 extends DrillHash{
+
+  public final class LongPair {
+    public long val1;
+    public long val2;
+  }
+
+  public static final long fmix64(long k) {
+    k ^= k >>> 33;
+    k *= 0xff51afd7ed558ccdL;
+    k ^= k >>> 33;
+    k *= 0xc4ceb9fe1a85ec53L;
+    k ^= k >>> 33;
+    return k;
+  }
+
+
+
+  /*
+  Take 64 bit of murmur3_128's output
+   */
+  public static long murmur3_64(long bStart, long bEnd, DrillBuf buffer, int seed) {
+
+    long h1 = seed & 0x00000000FFFFFFFFL;
+    long h2 = seed & 0x00000000FFFFFFFFL;
+
+    final long c1 = 0x87c37b91114253d5L;
+    final long c2 = 0x4cf5ad432745937fL;
+    long start = buffer.memoryAddress() + bStart;
+    long end = buffer.memoryAddress() + bEnd;
+    long length = end - start;
+    long roundedEnd = start + ( length & 0xFFFFFFF0);  // round down to 16 byte block
+    for (long i=start; i<roundedEnd; i+=16) {
+      long k1 = getLongLittleEndian(i);
+      long k2 = getLongLittleEndian(i+8);
+      k1 *= c1; k1  = Long.rotateLeft(k1,31); k1 *= c2; h1 ^= k1;
+      h1 = Long.rotateLeft(h1,27); h1 += h2; h1 = h1*5+0x52dce729;
+      k2 *= c2; k2  = Long.rotateLeft(k2,33); k2 *= c1; h2 ^= k2;
+      h2 = Long.rotateLeft(h2,31); h2 += h1; h2 = h2*5+0x38495ab5;
+    }
+
+    long k1 = 0;
+    long k2 = 0;
+
+    // tail
+    switch ((int)length & 15) {
+      case 15: k2  = (PlatformDependent.getByte(roundedEnd+14) & 0xffL) << 48;
+      case 14: k2 |= (PlatformDependent.getByte(roundedEnd+13) & 0xffL) << 40;
```

Comment:
Not particular reason. I was following the concept behind the code which was taking in rest of input as two long(k1 and k2) so OR is better conveying this concept. Will keep consistent with original code.